modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -73,87 +68,133 @@ function firstPass(features) {
 
           // eslint-disable-next-line max-depth
           for (const coord of flatten(polygon)) {
-            // eslint-disable-next-line max-depth
-            if (coord.length === 3) maxCoordLength = 3;
+            coordLengths.add(coord.length);
           }
         }
         break;
       default:
         throw new Error(`Unsupported geometry type: ${geometry.type}`);
     }
+
+    if (feature.properties) {
+      for (const key in feature.properties) {
+        const val = feature.properties[key];
+        numericProps[key] = numericProps[key] ? isNumeric(val) : numericProps[key];
+      }
+    }
   }
 
   return {
     pointPositions,
     linePositions,
     linePaths,
-    coordLength: maxCoordLength,
+    coordLength: Math.max(...coordLengths),
     polygonPositions,
     polygonObjects,
-    polygonRings
+    polygonRings,
+    // Array of keys whose values are always numeric
+    numericProps: Object.keys(numericProps).filter(k => numericProps[k])
   };
 }
 
 // Second scan over GeoJSON features
 // Fills coordinates into pre-allocated typed arrays
-function secondPass(features, firstPassData, options = {}) {
+function secondPass(features, options = {}) {
   const {
     pointPositions,
     linePositions,
     linePaths,
     coordLength,
     polygonPositions,
     polygonObjects,
-    polygonRings
-  } = firstPassData;
-  const {PositionDataType = Float32Array} = options;
+    polygonRings,
+    numericProps,
+    PositionDataType = Float32Array
+  } = options;
   const points = {
     positions: new PositionDataType(pointPositions * coordLength),
-    objectIds: new Uint32Array(pointPositions)
+    globalFeatureIndex: new Uint32Array(pointPositions),
+    featureIndex: new Uint32Array(pointPositions),
+    numericProps: {},
+    properties: []
   };
   const lines = {
-    pathIndices: new Uint32Array(linePaths),
+    pathIndices: new Uint32Array(linePaths + 1),
     positions: new PositionDataType(linePositions * coordLength),
-    objectIds: new Uint32Array(linePositions)
+    globalFeatureIndex: new Uint32Array(linePositions),
+    featureIndex: new Uint32Array(linePositions),
+    numericProps: {},
+    properties: []
   };
   const polygons = {
-    polygonIndices: new Uint32Array(polygonObjects),
-    primitivePolygonIndices: new Uint32Array(polygonRings),
+    polygonIndices: new Uint32Array(polygonObjects + 1),
+    primitivePolygonIndices: new Uint32Array(polygonRings + 1),
     positions: new PositionDataType(polygonPositions * coordLength),
-    objectIds: new Uint32Array(polygonPositions)
+    globalFeatureIndex: new Uint32Array(polygonPositions),
```

Comment:
For consistency as other arrays are named with plurals, these would be either `...Indices` or `...Ids`?